Data/test.lua

```diff
@@ -0,0 +1,9646 @@
+          Name="Bard ~ Versatile Performance",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_VersatilePerformance"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="2",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Well-Versed",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_WellVersed"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="5",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Lore Master",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_LoreMaster"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="10",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Jack of All Trades",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_JackOfAllTrades"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+  },
+  Name="Bard",
+  SpellStat="CHA",
+  Definitions={
+    {
+      Name="BardLVL",
+      InitialValue=0,
+    },
+    {
+      Name="CasterLevelBLBard",
+      InitialValue=0,
+    },
+  },
+  Bonuses={
+    {
+      Category="COMBAT",
+      Variables={
+        "BASEAB",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")*3/4",
+      Conditions={
+        function (character)
+          return (character.Variables["UseAlternateBABProgression"] == 0)
+        end,
+      },
+    },
+    {
+      Category="SAVE",
+      Variables={
+        "BASE.Reflex",
+        "BASE.Will",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")/2+2",
+      Conditions={
+        function (character)
+          return (character.Variables["UseAlternateSaveProgression"] == 0)
+        end,
+      },
+    },
+    {
+      Category="SAVE",
+      Variables={
+        "BASE.Fortitude",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")/3",
+      Conditions={
+        function (character)
+          return (character.Variables["UseAlternateSaveProgression"] == 0)
+        end,
+      },
+    },
+    {
+      Category="VAR",
+      Variables={
+        "ClassBABModerate",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")",
+      Conditions={
+        function (character)
+          return (character.Variables["UseFractionalBAB"] == 1)
+        end,
+      },
+    },
+    {
+      Category="VAR",
```

Comment:
Hrm. This seems like another one of those redundant things that could be inferred... every class having to say that X class has XLVL = CL.  I can't imagine a case where this _wouldn't_ be true.

Reply:
Its not always the same though, some classes count as other class levels, thats what these variables are for.

Reply:
Weird.  I guess like the magic counts as fighter level for feats and stuff.  I would imagine everything at _least_ counts as it's own level, but might count as some other levels too. :-)